src/pages/Fox/components/FoxFarming.tsx

```diff
@@ -0,0 +1,365 @@
+      }),
+    [assetAccountId],
+  )
+
+  const stakingOpportunityByIdFilter = useMemo(() => {
+    const userStakingId = assetAccountId
+      ? serializeUserStakingId(assetAccountId, opportunityId)
+      : undefined
+    return { userStakingId, opportunityId }
+  }, [assetAccountId, opportunityId])
+
+  const opportunity = useAppSelector(state =>
+    selectEarnUserStakingOpportunityByOpportunityId(state, stakingOpportunityByIdFilter),
+  )
+
+  const { getPeriodFinish } = useFoxFarming(ETH_FOX_STAKING_EVERGREEN_CONTRACT)
+
+  const {
+    data: periodFinish,
+    isLoading: isGetPeriodFinishQueryLoading,
+    isFetching: isGetPeriodFetchingLoading,
+  } = useQuery({
+    queryKey: ['getPeriodFinish'],
+    queryFn: () => getPeriodFinish(),
+  })
+
+  const isGetPeriodFinishLoading = useMemo(
+    () => isGetPeriodFinishQueryLoading || isGetPeriodFetchingLoading,
+    [isGetPeriodFinishQueryLoading, isGetPeriodFetchingLoading],
+  )
+
+  const fetchOpportunityData = useCallback(async () => {
+    if (!foxEthMarketData) return
+
+    const chainId = assetAccountId ? fromAccountId(assetAccountId).chainId : ethChainId
+
+    await fetchAllOpportunitiesIdsByChainId(appDispatch, chainId)
+    appDispatch(
+      opportunitiesApi.endpoints.getOpportunityMetadata.initiate(
+        [
+          {
+            opportunityId,
+            defiType: DefiType.Staking,
+            defiProvider: DefiProvider.EthFoxStaking,
+          },
+        ],
+        { forceRefetch: true },
+      ),
+    )
+
+    if (assetAccountId) {
+      appDispatch(
+        opportunitiesApi.endpoints.getOpportunityUserData.initiate(
+          [
+            {
+              opportunityId,
+              accountId: assetAccountId,
+              defiType: DefiType.Staking,
+              defiProvider: DefiProvider.EthFoxStaking,
+            },
+          ],
+          { forceRefetch: true },
+        ),
+      )
+    }
+
+    return true
+  }, [assetAccountId, appDispatch, foxEthMarketData, opportunityId])
+
+  useEffect(() => {
+    queryClient.invalidateQueries({
+      queryKey: ['fetchOpportunityData'],
+    })
+  }, [assetAccountId, queryClient])
+
+  const { isLoading: isOpportunityQueryLoading, isFetching: isOpportunityFetching } = useQuery({
+    queryKey: ['fetchOpportunityData', assetAccountId],
+    queryFn: fetchOpportunityData,
+    enabled: Boolean(foxEthMarketData),
+  })
+
+  const isOpportunityLoading = useMemo(
+    () => isOpportunityQueryLoading || isOpportunityFetching,
+    [isOpportunityQueryLoading, isOpportunityFetching],
+  )
+
+  const underlyingAsset = useAppSelector(state =>
+    selectAssetById(state, opportunity?.underlyingAssetId ?? ''),
+  )
+
+  const rewardAsset = useAppSelector(state =>
+    selectAssetById(state, opportunity?.rewardAssetIds[0] ?? ''),
+  )
+
+  const handleClick = useCallback(
+    (opportunity: LpEarnOpportunityType, action: DefiAction) => {
```

Comment:
```suggestion
    (opportunity: StakingEarnOpportunityType, action: DefiAction) => {
```

Reply:
Nice catch, done in [527f86c](https://github.com/shapeshift/web/pull/7934/commits/527f86ce55071bc81f46808931a053d3e5ee04e3)